From d22c37fe71eff6de6616e99907cd90f3890eaeef Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Fri, 12 Feb 2021 02:16:01 +0300
Subject: [PATCH 09/18] plat: renesas: Workaround bakery_lock build issue

This attempts to workaround the bakery lock build
issues when USE_COHERENT_MEM is disabled (CA76).

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 plat/renesas/rcar/include/platform_def.h |  2 +-
 plat/renesas/rcar/include/rcar_private.h | 19 ++++---------------
 2 files changed, 5 insertions(+), 16 deletions(-)

diff --git a/plat/renesas/rcar/include/platform_def.h b/plat/renesas/rcar/include/platform_def.h
index 755b176..563b3aa 100644
--- a/plat/renesas/rcar/include/platform_def.h
+++ b/plat/renesas/rcar/include/platform_def.h
@@ -195,7 +195,7 @@
  * per-cpu data structure for the RCAR port.
  ******************************************************************************/
 #if !USE_COHERENT_MEM
-#define PLAT_PCPU_DATA_SIZE	(2)
+#define PLAT_PCPU_DATA_SIZE	(4)
 #endif
 
 #endif /* PLATFORM_DEF_H */
diff --git a/plat/renesas/rcar/include/rcar_private.h b/plat/renesas/rcar/include/rcar_private.h
index a76c023..0a21aa4 100644
--- a/plat/renesas/rcar/include/rcar_private.h
+++ b/plat/renesas/rcar/include/rcar_private.h
@@ -35,6 +35,7 @@ typedef struct bl2_to_bl31_params_mem {
 #define rcar_lock_get()		bakery_lock_get(&rcar_lock)
 #define rcar_lock_release()	bakery_lock_release(&rcar_lock)
 #else
+#define RCAR_INSTANTIATE_LOCK	DEFINE_BAKERY_LOCK(rcar_lock);
 /*
  * Constants to specify how many bakery locks this platform implements. These
  * are used if the platform chooses not to use coherent memory for bakery lock
@@ -53,22 +54,10 @@ typedef struct rcar_cpu_data {
 	bakery_info_t pcpu_bakery_info[RCAR_MAX_BAKERIES];
 } rcar_cpu_data_t;
 
-#define RCAR_CPU_DATA_LOCK_OFFSET	\
-	__builtin_offsetof(rcar_cpu_data_t, pcpu_bakery_info)
-/*
- * Helper macros for bakery lock api when using the above rcar_cpu_data_t for
- * bakery lock data structures. It assumes that the bakery_info is at the
- * beginning of the platform specific per-cpu data.
- */
-#define rcar_lock_init(_lock_arg)
-
-#define rcar_lock_get(_lock_arg) 					\
-	bakery_lock_get(_lock_arg, 					\
-		CPU_DATA_PLAT_PCPU_OFFSET + RCAR_CPU_DATA_LOCK_OFFSET)
+#define rcar_lock_init()	bakery_lock_init(&rcar_lock)
+#define rcar_lock_get()		bakery_lock_get(&rcar_lock)
+#define rcar_lock_release()	bakery_lock_release(&rcar_lock)
 
-#define rcar_lock_release(_lock_arg)					\
-	bakery_lock_release(_lock_arg,	    				\
-		CPU_DATA_PLAT_PCPU_OFFSET + RCAR_CPU_DATA_LOCK_OFFSET)
 /* Ensure that the size of the RCAR specific per-cpu data structure and the size
  * of the memory allocated in generic per-cpu data for the platform are the same
  */
-- 
2.7.4

